homeworks/09-node-chokidar/modules/importer.ts

```diff
@@ -0,0 +1,44 @@
+import EventEmitter from 'events';
+import { promises, readFileSync } from 'fs';
+import { csvToJson } from 'csvtojson-converter';
+
+import { CHANGED_EVENT } from './dirWatcher';
+
+export const IMPORTED_EVENT = 'importer:imported';
+
+export class Importer {
+  constructor(public eventEmitter: EventEmitter, public dirPath: string) {}
+
+  listen = () => {
+    this.eventEmitter.addListener(CHANGED_EVENT, (fileNames) => {
+      if ((fileNames as string[]).length > 0) {
+        (fileNames as string[]).forEach((name) => {
+          const importedObjects = this.importSync(`${this.dirPath}/${name}`);
+          if (importedObjects.length > 0) {
+            this.eventEmitter.emit(IMPORTED_EVENT, importedObjects);
+          }
+        });
+      }
+    });
+  }
+
+  private import = async (path: string): Promise<Object[]> => (
+    promises.readFile(path)
+      .then((data) => csvToJson(data))
+      .catch(() => {
+        console.log(`File removed at ${path}`);
+        return [];
+      })
+  );
+
+  private importSync = (path: string): Object[] => {
+    let data;
+    try {
+      data = readFileSync(path, 'utf8');
```

Comment:
i guess that defining variable `data` should be inside try block

Reply:
move in https://github.com/vbarbasins/evo-ts-bootcamp/pull/10/commits/80a2d5181d37a9ab51cc552a661b6dbe7c257fb9